lib/mojito.js

```diff
@@ -907,78 +907,29 @@ Mojito = (function () {
         }
     };
 
-    // Adapted from https://github.com/requirejs/domReady
     Utils.domReady = (function () {
-        var readyCalls = [], isPageLoaded = false, scrollIntervalId,
-            testDiv, isTop;
+        var fns = [],
+            isPageLoaded = document.readyState === 'interactive' || document.readyState === 'complete' || document.readyState === 'loaded',
+            listener;
 
-        // Call domReady event callbacks one by one
         function callReady () {
-            var callbacks = readyCalls;
-
-            if (isPageLoaded) {
-                if (callbacks.length) {
-                    var i;
-                    for (i = 0; i < readyCalls.length; i += 1) {
-                        readyCalls[i]();
-                    }
-                    readyCalls = [];
-                }
-            }
-        }
-
-        // Detect when page is loaded
-        function pageLoaded () {
-            if (!isPageLoaded) {
-                isPageLoaded = true;
-                if (scrollIntervalId) {
-                    clearInterval(scrollIntervalId);
-                }
-
-                callReady();
-            }
-        }
-
-        if (document.addEventListener) {
-            document.addEventListener('DOMContentLoaded', pageLoaded, false);
-            window.addEventListener('load', pageLoaded, false);
-        } else if (window.attachEvent) {
-            document.attachEvent('onreadystatechange', function () {
-                if (document.readyState == 'interactive' || document.readyState == 'complete' || document.readyState == 'loaded') {
-                    pageLoaded('readaySate');
-                }
-            });
-
-            window.attachEvent('onload', pageLoaded);
-
-            testDiv = document.documentElement;
-            try {
-                isTop = window.frameElement === null;
-            } catch (e) {
-            }
-
-            if (testDiv.doScroll && isTop && window.external) {
-                scrollIntervalId = setInterval(function () {
-                    try {
-                        testDiv.doScroll();
-                        pageLoaded('ie hack');
-                    } catch (e) {
-                    }
-                }, 2);
+            isPageLoaded = true;
+            while (fns.length) {
+                listener = fns.shift();
```

Comment:
We can maybe save a couple of lines here (and remove the declaration for `listener`).
```
while (fns.length) {
    fns.shift().call();
}

Reply:
@dapperdrop 
Good suggestion, I will change it.

Reply:
@dapperdrop 
Fixed, but I changed it like this:

```
while (fns.length) {
 fns.shift()();
}
```

Because  `fns.shift().call()` changes the `this` reference to undefined, we have `this.activate = this.activate.bind(this);``` code in Mojito lib to ensure the `this` reference always being the test object.

Reply:
Cool - that makes sense. Thanks @allmywant